morpheus/_lib/tests/test_deserializers.cpp

```diff
@@ -0,0 +1,55 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "./test_morpheus.hpp"  // IWYU pragma: associated
+
+#include "morpheus/io/deserializers.hpp"
+
+#include <cudf/io/types.hpp>
+#include <gtest/gtest.h>
+
+#include <filesystem>
+#include <string>
+#include <vector>
+
+using namespace morpheus;
+
+TEST_CLASS(Deserializers);
+
+TEST_F(TestDeserializers, GetIndexColCount)
```

Comment:
We should also add tests here that verify round trip consistency. That we can read then write a file exactly as it was originally.